Cubical/Data/Int/MoreInts/DiffInt/Properties.agda

```diff
@@ -52,3 +56,172 @@ private
+                                                            fst b +ℕ snd a ≡⟨ sym r ⟩
+                                                            fst a +ℕ snd b ≡⟨ ℕ.+-comm (fst a) (snd b) ⟩
+                                                            snd b +ℕ fst a ∎
+  -ℤ' squash/ a a₁ p q i i₁ = squash/ (-ℤ' a) (-ℤ' a₁) (cong (λ x → -ℤ' x) p) (cong (λ x → -ℤ' x) q) i i₁
+
+_-_ : ℤ → ℤ → ℤ
+a - b = a + (-ℤ' b)
+
+open import Cubical.Algebra.Group.Base
+open import Cubical.Algebra.Monoid.Base
+open import Cubical.Algebra.Semigroup
+
+private
+
+  -ℤ'-invʳ-lem : (b : ℤ) → (Σ (ℕ × ℕ) ( λ (c , d) → [ (c , d) ]  ≡ b )) → b + (-ℤ' b) ≡ [ 0 , 0 ]
+  -ℤ'-invʳ-lem b ( (c , d) , p ) =
+    b + (-ℤ' b)
+      ≡⟨ cong (λ x → x + (-ℤ' b)) (sym p) ⟩
+    [ c , d ] + (-ℤ' b)
+      ≡⟨ cong (λ x → [ c , d ] + (-ℤ' x)) (sym p) ⟩
+    [ c +ℕ d , d +ℕ c ]
+      ≡⟨ cong (λ x → [ c +ℕ d , x ]) (ℕ.+-comm d c) ⟩
+    [ c +ℕ d , c +ℕ d ]
+      ≡⟨ ℤ-cancelˡ c ⟩
+    [ d -ℕ' d ]
+      ≡⟨ sym(ℤ-cancelʳ 0) ⟩
+    [ d +ℕ 0 -ℕ' d +ℕ 0 ]
+      ≡⟨ ℤ-cancelˡ d ⟩
+    [ 0 , 0 ] ∎
+
+  -ℤ'-invˡ-lem : (b : ℤ) → (Σ (ℕ × ℕ) ( λ (c , d) → [ (c , d) ]  ≡ b )) → (-ℤ' b) + b ≡ [ 0 , 0 ]
+  -ℤ'-invˡ-lem b ( (c , d) , p ) =
+    (-ℤ' b) + b
+      ≡⟨ cong (λ x → (-ℤ' b) + x) (sym p) ⟩
+    (-ℤ' b) + [ c , d ]
+      ≡⟨ cong (λ x → (-ℤ' x) + [ c , d ]) (sym p) ⟩
+    [ d +ℕ c , c +ℕ d ]
+      ≡⟨ cong (λ x → [ x , c +ℕ d ]) (ℕ.+-comm d c) ⟩
+    [ c +ℕ d , c +ℕ d ]
+      ≡⟨ ℤ-cancelˡ c ⟩
+    [ d -ℕ' d ]
+      ≡⟨ sym(ℤ-cancelʳ 0) ⟩
+    [ d +ℕ 0 -ℕ' d +ℕ 0 ]
+      ≡⟨ ℤ-cancelˡ d ⟩
+    [ 0 , 0 ] ∎
+
+-ℤ'-invʳ : (b : ℤ) → b + (-ℤ' b) ≡ [ 0 , 0 ]
+-ℤ'-invʳ b = PropositionalTruncation.rec (lem2 b) (-ℤ'-invʳ-lem b) ([]surjective b)
+                                               where
+                                                 lem2 : (b : ℤ) → isProp (b + (-ℤ' b) ≡ [ 0 , 0 ])
+                                                 lem2 b = ℤ-isSet (b + (-ℤ' b)) [ 0 , 0 ]
+
+-ℤ'-invˡ : (b : ℤ) → (-ℤ' b) + b ≡ [ 0 , 0 ]
+-ℤ'-invˡ b = PropositionalTruncation.rec (lem2 b) (-ℤ'-invˡ-lem b) ([]surjective b)
+                                               where
+                                                 lem2 : (b : ℤ) → isProp ((-ℤ' b) + b ≡ [ 0 , 0 ])
+                                                 lem2 b = ℤ-isSet ((-ℤ' b) + b) [ 0 , 0 ]
+
+ℤ+-assoc : ∀ x y z → x + (y + z) ≡ (x + y) + z
+ℤ+-assoc = elimProp3 (λ _ _ _ → ℤ-isSet _ _)
+  (λ { (a , b) (c , d) (e , f) i → [ ℕ.+-assoc a c e i -ℕ' ℕ.+-assoc b d f i ] })
+
+ℤ-isGroup : IsGroup {G = ℤ} [ 0 , 0 ] _+_ -ℤ'_
+IsSemigroup.is-set (IsMonoid.isSemigroup (IsGroup.isMonoid ℤ-isGroup)) = ℤ-isSet
+IsSemigroup.assoc (IsMonoid.isSemigroup (IsGroup.isMonoid ℤ-isGroup)) = ℤ+-assoc
+IsMonoid.identity (IsGroup.isMonoid ℤ-isGroup) = λ x → (zero-identityʳ 0 x  , zero-identityˡ 0 x)
+IsGroup.inverse ℤ-isGroup = λ x → (-ℤ'-invʳ x , -ℤ'-invˡ x)
+
+Int = group ℤ [ 0 , 0 ] _+_ -ℤ'_ ℤ-isGroup
```

Comment:
If you add the type, i.e. 'Group ℓ-zero' it will be easier for a user to spot, where the actual group is defined.
Also, if you like, you could add a prove that the multiplication is commutative and turn it into an 'AbGroup'.

Reply:
Thanks for the comment,
I have fixed it now.